src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java

```
@@ -679,7 +682,15 @@ public void provideCrafting(ICraftingProviderHelper craftingTracker) {
 
         for (PatternSlot slot : internalInventory) {
             if (slot == null) continue;
-            craftingTracker.addCraftingOption(this, slot.getPatternDetails());
+            ICraftingPatternDetails details = slot.getPatternDetails();
+            if (details == null) {
+                logger.warn(
+                    "Found an invalid pattern at " + getBaseMetaTileEntity().getCoords()
```

might cause spam, let me know if needed (should it just silently ignore?)

---

Use `GT_Mod.GT_FML_LOGGER` for logger